lib/resty/acme/client.lua

```diff
@@ -44,18 +44,24 @@ local function new_httpc()
   return httpc
 end
 
+local function set_account_key(self, account_key)
+  local account_pkey = openssl.pkey.new(account_key)
+  self.account_pkey = account_pkey
+  local account_thumbprint, err = util.thumbprint(account_pkey)
+  if err then
+    return "failed to calculate thumbprint: " .. err
```

Comment:
let's use a `return_value, error_msg` pattern here (also we need to update from the caller side)
with this function it will be `return false, <err>` on error and `return true` on success.